src/NServiceBus.Core.Tests/Msmq/MsmqSubscriptionStorageQueueTests.cs

```diff
@@ -13,32 +14,60 @@
     public class MsmqSubscriptionStorageQueueTests
     {
         [Test]
-        public async Task Subscribe_and_unsubscribe_is_persistent()
+        public async Task Subscribe_is_persistent()
         {
             var queue = new FakeStorageQueue();
+            var messageType = new MessageType(typeof(SomeMessage));
             var storage = CreateAndInit(queue);
 
-            var messageType = new MessageType(typeof(SomeMessage));
-            var messageTypes = new[] {messageType};
             await storage.Subscribe(new Subscriber("sub1", null), messageType, new ContextBag());
+            await storage.Subscribe(new Subscriber("sub2", "endpointA"), messageType, new ContextBag());
+
+            var storedMessages = queue.GetAllMessages().ToArray();
+            Assert.That(storedMessages.Length, Is.EqualTo(2));
```

Comment:
We are mixing Arrange and Asserts here. Is this needed? Since it seems to assert against `FakeStorageQueue`